[Bug] [TPartMin] pto.vbr使用 pto.PadValue.MAX.eval(dtype)后报错：loc("-":26:28): error: use of value '%pad_scalar_8' expects different type than prior uses: 'ui16' vs 'i16'

### Component

PTO Dialect / ODS (include/PTO/IR)

### Description

<img width="1528" height="1164" alt="Image" src="https://github.com/user-attachments/assets/ccb43bc9-4830-4b2f-95d8-2d6949247777" />

### Reproduction (minimal)

```markdown
运行命令：


python test/tilelang_st/script/run_st.py -r sim -v a5 -t tpartmin -c f32_64x64_full


Template文件：

"""TileLang DSL template for pto.tpartmin"""

import tilelang_dsl as pto


@pto.vkernel(
    target="a5",
    op="pto.tpartmin",
    advanced=True,
)
def template_tpartmin(src0: pto.Tile, src1: pto.Tile, dst: pto.Tile):
    dtype = dst.element_type
    valid_rows, valid_cols = dst.valid_shape
    src0_valid_rows, src0_valid_cols = src0.valid_shape
    src1_valid_rows, src1_valid_cols = src1.valid_shape
    lanes = pto.get_lanes(dtype)

    pad_scalar = pto.PadValue.MAX.eval(dtype)
    pad_vec = pto.vbr(pad_scalar)

    for row in range(0, valid_rows, 1):
        remained = valid_cols
        for col in range(0, valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            # pto.vsts(pad_vec, dst[row, col:], mask)
            pass

    for row in range(0, src0_valid_rows, 1):
        remained = src0_valid_cols
        for col in range(0, src0_valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            vec0 = pto.vlds(src0[row, col:])
            pto.vsts(vec0, dst[row, col:], mask)

    for row in range(0, src1_valid_rows, 1):
        remained = src1_valid_cols
        for col in range(0, src1_valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            vec_dst = pto.vlds(dst[row, col:])
            vec1 = pto.vlds(src1[row, col:])
            result = pto.vmin(vec_dst, vec1, mask)
            pto.vsts(result, dst[row, col:], mask)

    return
```

### Expected behavior

编译运行通过

### Actual behavior / error logs

```text
expand_helper: warning: failed to import
  /home/caojian/workspace/git/github/mouliangyu/PTOAS/lib/TileOps/tmrgsort_template.py:
  parameter 'exhausted' must not declare a default value in TileLang DSL v1

  loc("-":26:28): error: use of value '%pad_scalar_8' expects different type
  than prior uses: 'ui16' vs 'i16'

  ExpandTileOp: failed to parse DSL output


  loc("/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/testcase/tpartmin/tpartmin.pto":532:5):
  error: ExpandTileOp: failed to instantiate tilelang template for tpartmin

  Error: backend lowering pass execution failed.


make[2]: *** [testcase/tpartmin/CMakeFiles/tpartmin_kernel.dir/build.make:85: testcase/tpartmin/tpartmin_kernel.ll] Error 1
make[2]: Leaving directory '/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/build'
make[1]: *** [CMakeFiles/Makefile2:114: testcase/tpartmin/CMakeFiles/tpartmin_kernel.dir/all] Error 2
make[1]: Leaving directory '/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/build'
make: *** [Makefile:91: all] Error 2

run failed: Command '['make', 'VERBOSE=1', '-j', '64']' returned non-zero exit status 2.
```

### Git commit

6ba227dd728a209274227d88fad9eabc70a36203

### Host platform

None

### Target Ascend arch (if relevant)

None

### PTOAS build level (if relevant)

None

--- a/lib/PTO/Transforms/PTOLowerToOpLibCalls.cpp
+++ b/lib/PTO/Transforms/PTOLowerToOpLibCalls.cpp
@@ -1,3 +1,11 @@
+// Copyright (c) 2026 Huawei Technologies Co., Ltd.
+// This program is free software, you can redistribute it and/or modify it under the terms and conditions of
+// CANN Open Software License Agreement Version 2.0 (the "License").
+// Please refer to the License for details. You may not use this file except in compliance with the License.
+// THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, EITHER EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT, MERCHANTABILITY, OR FITNESS FOR A PARTICULAR PURPOSE.
+// See LICENSE in the root of the software repository for the full text of the License.
+
 #include "PTO/IR/PTO.h"
 
 #include "PTOLowerToOpLibCalls.h"
@@ -201,7 +209,15 @@ FailureOr<bool> mlir::pto::tryCloneOpLibInlineBridgeOp(OpBuilder &builder,
     if (!mappedSrc)
       return failure();
 
-    mapping.map(cast.getResult(0), mappedSrc);
+    Type dstTy = cast.getResult(0).getType();
+    if (mappedSrc.getType() == dstTy) {
+      mapping.map(cast.getResult(0), mappedSrc);
+      return true;
+    }
+
+    auto clonedCast = builder.create<UnrealizedConversionCastOp>(
+        cast.getLoc(), TypeRange{dstTy}, ValueRange{mappedSrc});
+    mapping.map(cast.getResult(0), clonedCast.getResult(0));
     return true;
   }
 

--- a/tilelang-dsl/python/tilelang_dsl/lowering.py
+++ b/tilelang-dsl/python/tilelang_dsl/lowering.py
@@ -2475,16 +2475,12 @@ class _AuthoringRenderer:
         if isinstance(expr, SemanticBindingRef):
             return env.get(expr.binding.name, _RenderedValue(expr.binding.ssa_name, expr.type))
         if isinstance(expr, SemanticLiteralExpr):
-            if desired_name is not None and into is not None:
-                into.append(
-                    self._indent(indent)
-                    + f"{desired_name} = arith.constant {self._format_constant(expr.value, expr.type)} : "
-                    f"{self._render_arith_constant_type(expr.type)}"
-                )
-                return _RenderedValue(name=desired_name, type=expr.type)
-            return _RenderedValue(
-                name=self._materialize_constant(expr.value, expr.type),
-                type=expr.type,
+            return self._lower_literal_expr(
+                expr.value,
+                expr.type,
+                indent=indent,
+                desired_name=desired_name,
+                into=into,
             )
         if isinstance(expr, SemanticSubscriptAccess):
             return self._lower_subscript_access(
@@ -3680,6 +3676,56 @@ class _AuthoringRenderer:
             f"{self._render_arith_constant_type(ty)}"
         )
         return name
+
+    def _signless_integer_scalar_type(self, ty: SemanticType) -> SemanticScalarType | None:
+        if not isinstance(ty, SemanticScalarType) or not is_integer_dtype(ty.dtype):
+            return None
+        signedness = integer_signedness(ty.dtype)
+        if signedness in {None, "signless"}:
+            return None
+        bitwidth = integer_bitwidth(ty.dtype)
+        if bitwidth not in {8, 16, 32, 64}:
+            raise NotImplementedError(
+                f"unsupported integer bitwidth {bitwidth!r} for signless literal lowering"
+            )
+        return SemanticScalarType(dtype=ScalarType(f"i{bitwidth}"))
+
+    def _lower_literal_expr(
+        self,
+        value: object,
+        ty: SemanticType,
+        *,
+        indent: int,
+        desired_name: str | None = None,
+        into: list[str] | None = None,
+    ) -> _RenderedValue:
+        raw_type = self._signless_integer_scalar_type(ty) or ty
+        if desired_name is not None and into is not None and raw_type == ty:
+            into.append(
+                self._indent(indent)
+                + f"{desired_name} = arith.constant {self._format_constant(value, ty)} : "
+                f"{self._render_arith_constant_type(ty)}"
+            )
+            return _RenderedValue(name=desired_name, type=ty)
+
+        if desired_name is not None and into is not None:
+            raw_name = self._new_temp()
+            into.append(
+                self._indent(indent)
+                + f"{raw_name} = arith.constant {self._format_constant(value, raw_type)} : "
+                f"{self._render_arith_constant_type(raw_type)}"
+            )
+            into.append(
+                self._indent(indent)
+                + f"{desired_name} = builtin.unrealized_conversion_cast {raw_name} : "
+                f"{self._render_type(raw_type)} to {self._render_type(ty)}"
+            )
+            return _RenderedValue(name=desired_name, type=ty)
+
+        return _RenderedValue(
+            name=self._materialize_constant(value, ty),
+            type=ty,
+        )
 
     def _constant_name(self, value: object, ty: SemanticType) -> str:
         if isinstance(ty, SemanticIndexType):

--- a/tilelang-dsl/tests/test_tilelang_dsl_v1.py
+++ b/tilelang-dsl/tests/test_tilelang_dsl_v1.py
@@ -2448,6 +2448,32 @@ class TileLangDSLDescriptorTests(unittest.TestCase):
         self.assertIn("arith.constant 4294967295 : i32", text)
         self.assertNotIn("arith.constant 4294967295 : ui32", text)
 
+    def test_unsigned_pad_value_eval_broadcast_bitcasts_signless_literal(self) -> None:
+        @pto.vkernel(op="tile_pad_value_ui16_vbr_unique", dtypes=[(pto.ui16,)], advanced=True)
+        def kernel(tile: pto.Tile):
+            scalar = tile.pad_value.eval()
+            vec = pto.vbr(scalar)
+            return None
+
+        specialized = kernel.specialize(
+            tile=pto.TileSpecialization(
+                shape=(8, 16),
+                memory_space=pto.MemorySpace.UB,
+                config=pto.TileConfig.from_mapping(
+                    {
+                        "pad_value": pto.PadValue.MAX,
+                    }
+                ),
+            )
+        )
+
+        text = specialized.mlir_text()
+        self.assertIn("dtype=ui16", text)
+        self.assertIn("arith.constant 65535 : i16", text)
+        self.assertIn("builtin.unrealized_conversion_cast", text)
+        self.assertIn(": i16 to ui16", text)
+        self.assertIn("pto.vbr", text)
+
 
     def test_make_mask_vlds_vsts_and_vector_families_lower_inside_strict_vecscope(self) -> None:
         @pto.vkernel(op="eltwise", dtypes=[(pto.f32, pto.f32)], advanced=True)

